Assets/Xbox Live/Scripts/UserProfile.cs

```diff
@@ -86,53 +86,113 @@ private void XboxLiveUserOnSignOutCompleted(object sender, SignOutCompletedEvent
+            && XboxLiveUserManager.Instance.UserForSingleUserMode.User != null
+            && !XboxLiveUserManager.Instance.UserForSingleUserMode.User.IsSignedIn && !this.SignInCalledOnce)
+        {
+            this.StartCoroutine(this.SignInAsync());
+        }
 
-        this.User = XboxLiveComponent.Instance.User;
-        TaskYieldInstruction<SignInResult> signInTask = this.User.SignInAsync().AsCoroutine();
-        yield return signInTask;
+        if (this.XboxLiveUser != null && this.XboxLiveUser.User != null && !this.XboxLiveUser.User.IsSignedIn && !this.SignInCalledOnce)
+        {
+            this.StartCoroutine(this.SignInAsync());
+        }
 
-        // Throw any exceptions if needed.
-        if (signInTask.Result.Status != SignInStatus.Success)
+        if (!this.SignInCalledOnce && !string.IsNullOrEmpty(this.SignInKeyName) && Input.GetKeyDown(this.SignInKeyName))
         {
-            throw new Exception("Sign in Failed");
+            this.StartCoroutine(this.InitializeXboxLiveUser());
         }
 
-        XboxLive.Instance.StatsManager.AddLocalUser(this.User);
-        var addLocalUserTask = XboxLive.Instance.SocialManager.AddLocalUser(this.User, SocialManagerExtraDetailLevel.PreferredColor).AsCoroutine();
-        yield return addLocalUserTask;
+    }
 
-        if (addLocalUserTask.Task.IsFaulted)
-        {
-            throw addLocalUserTask.Task.Exception;
+    public IEnumerator InitializeXboxLiveUser()
+    {
+        yield return null;
+        // Disable the sign-in button
+        this.signInPanel.GetComponentInChildren<Button>().interactable = false;
+
+#if NETFX_CORE
+        if (!XboxLiveUserManager.Instance.SingleUserModeEnabled) {
+            var autoPicker = new Windows.System.UserPicker { AllowGuestAccounts = this.AllowGuestAccounts};
+            autoPicker.PickSingleUserAsync().AsTask().ContinueWith(
+                    task =>
+                        {
+                            if (task.Status == TaskStatus.RanToCompletion)
+                            {
+                                this.XboxLiveUser.WindowsSystemUser = task.Result;
+                                this.XboxLiveUser.Initialize();
+                            }
+                            else
+                            {
+                                Debug.Log("Exception occured: "+ task.Exception.Message);
+                            }
+                        });
+        } 
+        else {
+           this.XboxLiveUser = XboxLiveUserManager.Instance.UserForSingleUserMode;
+           this.XboxLiveUser.Initialize();
+        }
+#else
+        if(this.XboxLiveUser == null){
+            this.XboxLiveUser = XboxLiveUserManager.Instance.GetSingleModeUser();
         }
 
-        yield return this.LoadProfileInfo();
+        this.XboxLiveUser.Initialize();
+#endif
+
+    }
+
+    public IEnumerator SignInAsync()
+    {
+        this.SignInCalledOnce = true;
```

Comment:
should this be set outside the coroutine?  aka is it possible for the Coroutine to not run immediately?